runtime/src/main/java/org/corfudb/protocols/logprotocol/CheckpointEntry.java

```diff
@@ -162,6 +162,12 @@ void deserializeBuffer(ByteBuf b, CorfuRuntime rt) {
     @Override
     public void serialize(ByteBuf b) {
         super.serialize(b);
+
+        if (cpType == CheckpointEntryType.END
+                && getDict().get(CheckpointDictKey.SNAPSHOT_ADDRESS) == null) {
+            throw new RuntimeException("CP end record must contain a snapshot address");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)